### PR TITLE
Created enumarable

### PR DESCRIPTION
**Pull request by alphayowakarindi**
At this juncture of development we implemented the following:
- Created a class and a method to yield methods from a module.
- Created an enumerable module with several methods to check for true or false status.
- Implemented the methods to yield.

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require_relative 'my_enumerable'
+
+class Mylist
+  include Enumerable
+
+  def initialize(*list)
+    @list = list
```
Comment (GzimAsani) [inline my_list.rb:7]: - [  ] I would suggest you add them this way, it is shorter and looks better and adds them to your list which is an array
```ruby
 def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  attr_reader :list
+
+  def each
+    yield @list
+  end
+end
```
Comment (GzimAsani) [inline my_list.rb]: - [ ] Your class needs to implement `#each` method that yields all members of `@list` in succession. Similar:

```ruby
def each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
end
```

File: my_enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module Enumerable
+    def all?
+      list.all? { |num| num < 5 }
+    end
+  
+    def any?
+      list.any? { |num| num == 2 }
+    end
```
Comment (GzimAsani) [inline my_enumerable.rb]: - [ ] You need to work again on the all and any, they are too specific, you need to take into consideration other cases where they should return false for example:
- list_of_nums.all? {|e| e > 5} should return false, but currently it returns true
- list_of_nums.any? {|e| e == 5} should return false, but currently it returns true

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello    @alphayowakarindi and @basitali111       :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your first ruby project, it looks great +1:  
 
 ## Required Changes 

- [ ] Please update your readme file, using the Microverse template
- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @alphayowakarindi 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 

 ## Required Changes ♻️
* As mentioned by the previous reviewer your readme file is not descriptive kindly Add a descriptive README file to your project - please use [this template](https://github.com/microverseinc/readme-template) but remember to customize it to your project.
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions
-N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- ### Please add linters, I forgot to mention on the firs review

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**